samples/drivers/adc/src/main.c

```diff
@@ -22,149 +29,58 @@ static const struct adc_dt_spec adc_channels[] = {
-		adc_channel_setup(dt_spec->dev, &dt_spec->channel_cfg);
-
-		/* For the internal reference, use the voltage value returned
-		 * by the dedicated API function. For others, use the value
-		 * from devicetree if available.
-		 */
-		if (dt_spec->channel_cfg.reference == ADC_REF_INTERNAL) {
-			*vref_mv = adc_ref_internal(dt_spec->dev);
-		} else if (dt_spec->vref_mv > 0) {
-			*vref_mv = dt_spec->vref_mv;
-		}
-	} else {
-		struct adc_channel_cfg channel_cfg = {
-			.channel_id       = dt_spec->channel_id,
-			.gain             = ADC_GAIN,
-			.reference        = ADC_REFERENCE,
-			.acquisition_time = ADC_ACQUISITION_TIME,
-		};
-
-		adc_channel_setup(dt_spec->dev, &channel_cfg);
-
-		*vref_mv = adc_ref_internal(dt_spec->dev);
-	}
-}
-
-static void prepare_sequence(struct adc_sequence *sequence,
-			     const struct adc_dt_spec *dt_spec)
-{
-	sequence->channels     = BIT(dt_spec->channel_id);
-	sequence->resolution   = ADC_RESOLUTION;
-	sequence->oversampling = 0;
-
-	if (dt_spec->channel_cfg_dt_node_exists) {
-		if (dt_spec->resolution) {
-			sequence->resolution = dt_spec->resolution;
-		}
-		if (dt_spec->oversampling) {
-			sequence->oversampling = dt_spec->oversampling;
-		}
-	}
-}
-
-static void print_millivolts(int32_t value,
-			     uint16_t vref_mv,
-			     uint8_t resolution,
-			     const struct adc_dt_spec *dt_spec)
-{
-	enum adc_gain gain = ADC_GAIN;
-
-	if (dt_spec->channel_cfg_dt_node_exists) {
-		gain = dt_spec->channel_cfg.gain;
-
-		/*
-		 * For differential channels, one bit less needs to be specified
-		 * for resolution to achieve correct conversion.
-		 */
-		if (dt_spec->channel_cfg.differential) {
-			resolution -= 1;
-		}
-	}
-
-	adc_raw_to_millivolts(vref_mv, gain, resolution, &value);
-	printk(" = %d mV", value);
-}
-
 void main(void)
 {
 	int err;
-	int16_t sample_buffer[1];
+	int16_t buf;
 	struct adc_sequence sequence = {
-		.buffer      = sample_buffer,
+		.buffer = &buf,
 		/* buffer size in bytes, not number of samples */
-		.buffer_size = sizeof(sample_buffer),
+		.buffer_size = sizeof(buf),
 	};
-	uint16_t vref_mv[ARRAY_SIZE(adc_channels)] = { 0 };
 
 	/* Configure channels individually prior to sampling. */
-	for (uint8_t i = 0; i < ARRAY_SIZE(adc_channels); i++) {
+	for (size_t i = 0U; i < ARRAY_SIZE(adc_channels); i++) {
 		if (!device_is_ready(adc_channels[i].dev)) {
-			printk("ADC device not found\n");
+			printk("ADC controller device not ready\n");
 			return;
 		}
 
-		configure_channel(&adc_channels[i], &vref_mv[i]);
+		err = adc_channel_setup_dt(&adc_channels[i]);
+		if (err < 0) {
+			printk("Could not setup channel #%d (%d)\n", i, err);
+			return;
+		}
 	}
 
 	while (1) {
 		printk("ADC reading:\n");
-		for (uint8_t i = 0; i < ARRAY_SIZE(adc_channels); i++) {
-			printk("- %s, channel %d: ",
-				adc_labels[i], adc_channels[i].channel_id);
+		for (size_t i = 0U; i < ARRAY_SIZE(adc_channels); i++) {
+			int32_t val_mv;
+
+			printk(" - %s, channel %d: ",
```

Comment:
```suggestion
			printk("- %s, channel %d: ",
```
See a78e7c5c90d8cb31ec550ce289503bd698675585.